### PR TITLE
[ES6 class]  Update MetadataStore to es6 class

### PR DESCRIPTION
**Pull request by orlandohohmeier**
---
ℹ️   _This branch depends on features that will be introduced with #275, thus it will fail until the respective branch is merged._

---
Update store to es6 class – this is part of series of PRs to update all the stores as discussed in https://github.com/dcos/dcos-ui/pull/246/files/b4b52fcdb5f899fd74f9bd68cfa3d216177fcecf#r64962494

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @Poltergeist

File: src/js/stores/MetadataStore.js [hunk 1]
```diff
@@ -1,78 +1,83 @@
...
+          // only emitting on change
+          if (!deepEqual(oldDCOSMetadata, dcosMetadata)) {
+            this.set({dcosMetadata});
+            this.emitChange(EventTypes.DCOS_METADATA_CHANGE);
+          }
+          break;
+      }
+
+      return true;
+    })
```
Comment (mlunoe) [inline src/js/stores/MetadataStore.js]: #nit missing semi-colon

**Comment by Poltergeist** [comment]
Code looks good but Integration tests is failing.

**Comment by orlandohohmeier** [comment]
retest this please

**Pull request merged**